Team54/src/main/java/ISA/Team54/users/controller/PharmacyController.java

```diff
@@ -77,6 +83,42 @@
 		
 		pharmacies.forEach(p -> pharmacyDTOs.add(new PharmacyMapper().PharmacyToPharmacyDTO(p)));
 		return new ResponseEntity<List<PharmacyDTO>>(pharmacyDTOs, HttpStatus.OK);
+	}
+
+	@GetMapping("/patient-pharmacies")
+	@PreAuthorize("hasRole('ROLE_PATIENT')")
+	public ResponseEntity<List<PharmacyDTO>> getPatientPharmacies(){
+		try {
+			List<Pharmacy> pharmacies = pharmacyService.getPatientPharmacies();
+			pharmacies.addAll(drugReservationService.getPatientPharmacies());
+			List<Pharmacy> distinctPharmacies = getPharmaciesWithoutDuplicates(pharmacies);
+			List<PharmacyDTO> pharmacyDTOs = new ArrayList<PharmacyDTO>();
+			distinctPharmacies.forEach(
+					p -> pharmacyDTOs.add(new PharmacyMapper().PharmacyToPharmacyDTO(p))
+			);
+			return new ResponseEntity<List<PharmacyDTO>>(pharmacyDTOs, HttpStatus.OK);
+		}catch (Exception e){
+			return new ResponseEntity<>(HttpStatus.INTERNAL_SERVER_ERROR);
+		}
+	}
+
+	private List<Pharmacy> getPharmaciesWithoutDuplicates(List<Pharmacy> pharmacies) {
+		List<Pharmacy> distinctPharmacies = new ArrayList<Pharmacy>();
+		pharmacies.forEach(
+				p -> {
+					if(CheckIfPharmacyUnique(p, distinctPharmacies))
+						distinctPharmacies.add(p);
+				}
+		);
+
+		return distinctPharmacies;
+	}
 
+	private boolean CheckIfPharmacyUnique(Pharmacy pharmacy, List<Pharmacy> pharmacies) {
+		for (Pharmacy p : pharmacies) {
+			if(p.getId() == pharmacy.getId())
+				return false;
+		}
+		return true;
```

Comment:
i ovo takodje

Reply:
Ni to ne moze zato sto taj endpoint dobavlja podatke iz vise njih i tu ih uvezuje.

Reply:
okej